### PR TITLE
Shipit config and updates

### PR DESCRIPTION
**Pull request by wizardlyhel**
### What are you trying to accomplish with this PR?

Allow Slate to be auto-deploy when merge to master

### Checklist
For contributors:
- [x] I have [updated the docs](https://github.com/Shopify/slate/blob/master/CONTRIBUTING.md#documentation) to reflect these changes, if applicable.

For maintainers:
- [ ] I have :tophat:'d these changes.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -112,60 +112,65 @@ You only need to update the source files and Travis CI will take care of deployi
...
+   This command will:
+
+   1. Run the equivalent of lerna updated to determine which packages need to be published.
+   2. If necessary, increment the version key in lerna.json.
+   3. Update the package.json of all updated packages to their new versions.
+   4. Update all dependencies of the updated packages with the new versions, specified with a caret (^).
+   5. Create a new git commit and tag for the new version.
+   6. Git push to origin master with the newly created tag.
+
+   _Note: See the [lerna documentation](https://github.com/lerna/lerna/tree/2.x#publish) for more information_
```
Comment (antoinegrant) [inline CONTRIBUTING.md]: These steps refer to lerna `2.x` docs. We would need to update those.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -112,60 +112,69 @@ You only need to update the source files and Travis CI will take care of deployi
...
+2. Checkout `master` and pull the latest from the origin
+
+   ```
+   slate $ git checkout master && git pull origin master
+   ```
+
+3. You can verify that there are indeed packages to be published (optional)
+
+   ```
+   slate $ yarn lerna updated
```
Comment (antoinegrant) [inline CONTRIBUTING.md]: Altho `lerna updated` still works, `lerna updated` is the official way. 

https://github.com/lerna/lerna/tree/master/commands/changed#readme

File: lerna.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "lerna": "2.0.0",
+  "lerna": "3.11.0",
   "version": "1.0.0-beta.17",
   "npmClient": "yarn",
   "useWorkspaces": true,
```
Comment (antoinegrant) [inline lerna.json]: You'll need to tell lerna to not publish to npm when running `lerna publish`. The actual publishing will happen in Shipit.

You'll need to add this:
https://github.com/Shopify/theme-scripts/blob/master/lerna.json#L9-L13

Comment (wizardlyhel) [reply lerna.json]: for `yarn release`, I change it from `yarn publish` to `yarn version`

Comment (antoinegrant) [reply lerna.json]: Ah right because we're on `3.x` now.

**Pull request merged**